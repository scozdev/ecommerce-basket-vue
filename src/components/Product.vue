<template>
  <div class="product">
    <img :src="productData.image" :alt="productData.name" />

    <span class="description">{{ productData.name }}</span>

    <div class="button-container">
      <span class="price">{{ productData.price }}</span>
      <Button @click="addCart(productData)" />
    </div>
  </div>
</template>

<script>
import Button from "@/components/Button.vue";

import { mapActions } from "vuex";

export default {
  name: "Product",
  components: {
    Button,
  },
  props: ["productData"],

  methods: {
    ...mapActions(["addItemToCart"]),
    addCart(product) {
      this.addItemToCart(product);
      this.$router.push(`/basket`);
    },
  },
};
</script>

<style scoped>
.product {
  border: 1px solid rgba(0, 0, 0, 0.1);
  padding: 10px 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  align-items: center;
}
.product > *:not(:last-child) {
  margin-bottom: 15px;
}
.description,
.price {
  color: rgba(0, 0, 0, 0.5);
}
img {
  width: 80%;
}

.button-container {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}
</style>