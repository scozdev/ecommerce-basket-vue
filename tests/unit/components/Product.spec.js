import { mount, RouterLinkStub, shallowMount } from '@vue/test-utils';
import Product from '@/components/Product';

describe('Product.vue', () => {
  let product;
  let wrapper;

  beforeEach(() => {
    product = {
      id: 1,
      name: 'Åžampuan',
      price: '13',
      currency: 'TRY',
      image:
        'https://cdn.glitch.com/a28552e7-44e1-4bbd-b298-5745e70c2209%2Fsampuan.jpeg?v=1561027551321',
    };

    wrapper = shallowMount(Product, {
      propsData: {
        productData: product,
      },
    });
  });

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('class property test', () => {
    expect(wrapper.classes()).toContain('product');
  });
});
