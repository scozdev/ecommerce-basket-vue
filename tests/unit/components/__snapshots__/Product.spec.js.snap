// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product.vue should match snapshot 1`] = `
<div class="product"><img src="https://cdn.glitch.com/a28552e7-44e1-4bbd-b298-5745e70c2209%2Fsampuan.jpeg?v=1561027551321" alt="Şampuan"><span class="description">Şampuan</span>
  <div class="button-container"><span class="price">13</span>
    <button-stub></button-stub>
  </div>
</div>
`;
